controllers/azurejson_machinepool_controller_test.go

```diff
@@ -210,3 +217,140 @@ func TestAzureJSONPoolReconciler(t *testing.T) {
 		})
 	}
 }
+
+func TestAzureJSONPoolReconcilerUserAssignedIdentities(t *testing.T) {
+	g := NewWithT(t)
+	ctrlr := gomock.NewController(t)
+	defer ctrlr.Finish()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-machine-pool", Namespace: "fake-ns"}}
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+
+	err := infrav1exp.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = expv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = corev1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = clusterv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = infrav1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	azureMP := &infrav1exp.AzureMachinePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-machine-pool",
+			Namespace: "fake-ns",
+			Labels: map[string]string{
+				clusterv1.ClusterNameLabel: "fake-cluster",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: expv1.GroupVersion.Group + "/" + expv1.GroupVersion.Version,
+					Kind:       "MachinePool",
+					Name:       "fake-other-machine-pool",
+					Controller: to.Ptr(true),
+				},
+			},
+		},
+		Spec: infrav1exp.AzureMachinePoolSpec{
+			UserAssignedIdentities: []infrav1.UserAssignedIdentity{
+				{
+					ProviderID: "fake-id",
+				},
+			},
+		},
+	}
+
+	clusterA := &clusterv1.Cluster{
```

Comment:
nit: I think naming this just `cluster` would be better since the "A" implies there's another cluster (or at least that's how I see it)

Reply:
I didn't want to shadow any variable but I can change this.

Reply:
It doesn't shadow, so I'll change this.

Reply:
Changed to cluster as suggested, resolving.